### PR TITLE
wl-gammarelay-applet: init at 0.1.4

### PR DESCRIPTION
**Pull request by lgbishop**
## Description of changes

wl-gammarelay-applet is a small desktop applet for controlling wl-gammarelay-rs via DBus. This applet is written in Rust and provides a Slint UI.

The original repo is at https://github.com/junelva/wl-gammarelay-applet
This package derivation uses the fork https://github.com/lgbishop/wl-gammarelay-applet which contains a few UI changes/fixes.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by lgbishop** [comment]
Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>wl-gammarelay-applet</li>
  </ul>
</details>

File: pkgs/by-name/wl/wl-gammarelay-applet/package.nix [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        lib.makeLibraryPath [
+          wayland
+          libxkbcommon
+          fontconfig
+        ]
+      }
+  '';
+
+  meta = {
+    description = "Control wl-gammarelay-rs via applet.";
```
Comment (Aleksanaa) [inline pkgs/by-name/wl/wl-gammarelay-applet/package.nix:54]:
```suggestion
    description = "Control wl-gammarelay-rs via applet";
```

File: pkgs/by-name/wl/wl-gammarelay-applet/package.nix [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    longDescription = ''
+      wl-gammarelay-applet is a small desktop applet for controlling wl-gammarelay-rs via DBus. This applet is written in Rust and provides a Slint UI.
```
Comment (Aleksanaa) [inline pkgs/by-name/wl/wl-gammarelay-applet/package.nix:56]:
```suggestion
      wl-gammarelay-applet is a small desktop applet for controlling
      wl-gammarelay-rs via DBus. This applet is written in Rust and
      provides a Slint UI.
```

File: pkgs/by-name/wl/wl-gammarelay-applet/package.nix [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  fetchFromGitHub,
+  rustPlatform,
+  stdenv,
+  wayland,
+  libxkbcommon,
+  fontconfig,
+  pkg-config,
+}:
+
+rustPlatform.buildRustPackage rec {
```
Comment (Aleksanaa) [inline pkgs/by-name/wl/wl-gammarelay-applet/package.nix:12]: unused `rec`?

File: pkgs/by-name/wl/wl-gammarelay-applet/package.nix [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    [
+    ]
+    ++ lib.optionals stdenv.isLinux [
+
+      fontconfig
+    ]
+    ++
+      lib.optionals stdenv.isDarwin
+        [
+        ];
```
Comment (Aleksanaa) [inline pkgs/by-name/wl/wl-gammarelay-applet/package.nix:40]:
```suggestion

  buildInputs = [
    fontconfig
  ];
```

File: pkgs/by-name/wl/wl-gammarelay-applet/package.nix [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+  postFixup = lib.optionalString stdenv.isLinux ''
+    patchelf $out/bin/wl-gammarelay-applet \
+      --add-rpath ${
+        lib.makeLibraryPath [
+          wayland
+          libxkbcommon
+          fontconfig
+        ]
+      }
+  '';
```
Comment (Aleksanaa) [inline pkgs/by-name/wl/wl-gammarelay-applet/package.nix:51]:
```suggestion
  runtimeDependencies = [
    wayland
    libxkbcommon
    fontconfig
  ];
```

Add `autoPatchelfHook` to `nativeBuildInputs`

File: pkgs/by-name/wl/wl-gammarelay-applet/package.nix [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+  src = fetchFromGitHub {
+    owner = "lgbishop";
+    repo = "wl-gammarelay-applet";
+    rev = "8a0d9e6364d7445fc69c59b2f168cfec91c2fe87";
+    sha256 = "sha256-t6bycmaquZ0IMs/WnAzkz5FnIWKIq0BTbeeoUFLeuYg=";
+  };
+
+  cargoLock = {
+    lockFile = ./Cargo.lock;
+  };
```
Comment (Aleksanaa) [inline pkgs/by-name/wl/wl-gammarelay-applet/package.nix:25]: Is their a particular reason to add `Cargo.lock`? Due to the current issues with repo size and eval overhead, we tend to use `cargoHash`, although this strategy may change in the future.

File: pkgs/by-name/wl/wl-gammarelay-applet/package.nix (hunk 3 above)
Comment (lgbishop) [reply pkgs/by-name/wl/wl-gammarelay-applet/package.nix]: Fixed: [https://github.com/lgbishop/nixpkgs/commit/7a04a5f084bf94aaefad20daaa32020fde3b0b36](url)

File: pkgs/by-name/wl/wl-gammarelay-applet/package.nix (hunk 5 above)
Comment (lgbishop) [reply pkgs/by-name/wl/wl-gammarelay-applet/package.nix]: Done: [https://github.com/lgbishop/nixpkgs/commit/a7f456d07e52b6b54fe655e0f6e003e2afe57331](url)

File: pkgs/by-name/wl/wl-gammarelay-applet/package.nix (hunk 6 above)
Comment (lgbishop) [reply pkgs/by-name/wl/wl-gammarelay-applet/package.nix]: No reason in particular - have changed it: [https://github.com/lgbishop/nixpkgs/commit/9b2a6631388462930c9a481512c45829c4c0f117](url)

**Review by Aleksanaa — APPROVED** [review]
Great! now please:

1. make sure it still works.
2. squash your commits into two:
  maintainers: add lgbishop
  wl-gammarelay-applet: init at 0.1.4

**Comment by lgbishop** [comment]
I believe I have done everything correctly:
1. Can confirm the output from `nix-build -A wl-gammarelay-applet` runs. Including the package as a derivation in my system flake also seems to work well on my personal machine.
2. I have squashed the commits as required.

@Aleksanaa thank you for your time and attention!

**Pull request merged**